modules/ppcp-fraud-protection/src/Recaptcha/Recaptcha.php

```diff
@@ -0,0 +1,561 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WooCommerce\PayPalCommerce\FraudProtection\Recaptcha;
+
+use Automattic\WooCommerce\Utilities\OrderUtil;
+use Psr\Log\LoggerInterface;
+use WC_Order;
+use WP_Error;
+
+class Recaptcha {
+	private const V2_CONTAINER_ID                = 'ppcp-recaptcha-v2-container';
+	private const ERROR_CODE_MISSING_TOKEN       = 'ppcp_recaptcha_missing_token';
+	private const ERROR_CODE_VERIFICATION_FAILED = 'ppcp_recaptcha_verification_failed';
+	private const CAPTCHA_USAGE_LIMIT            = 5;
+	private const CAPTCHA_RESULT_TRANSIENT_KEY   = 'ppcp_recaptcha_result_';
+	private const CAPTCHA_RESULT_META_KEY        = 'ppcp_recaptcha_captcha_result';
+
+	private RecaptchaIntegration $integration;
+
+	/**
+	 * The methods that require captcha.
+	 *
+	 * @var string[]
+	 */
+	private array $payment_methods;
+
+	private string $module_url;
+
+	private string $asset_version;
+
+	private LoggerInterface $logger;
+
+	/**
+	 * @param RecaptchaIntegration $integration
+	 * @param string[]             $payment_methods The methods that require captcha.
+	 * @param string               $module_url
+	 * @param string               $asset_version
+	 * @param LoggerInterface      $logger
+	 */
+	public function __construct(
+		RecaptchaIntegration $integration,
+		array $payment_methods,
+		string $module_url,
+		string $asset_version,
+		LoggerInterface $logger
+	) {
+
+		$this->integration     = $integration;
+		$this->payment_methods = $payment_methods;
+		$this->module_url      = $module_url;
+		$this->asset_version   = $asset_version;
+		$this->logger          = $logger;
+	}
+
+	protected function should_use_recaptcha(): bool {
+		if ( ! wc_string_to_bool( $this->integration->enabled ) ) {
+			return false;
+		}
+
+		if ( wc_string_to_bool( 'guest_only' ) && is_user_logged_in() ) {
```

Comment:
I think this should be `$this->integration->get_option( 'guest_only' )` instead of the hardcoded string `guest_only`, doesn't it?

Reply:
oops, yes :D